pkg/ipamd/ipamd.go

```diff
@@ -761,6 +797,11 @@ func (c *IPAMContext) tryAllocateENI(ctx context.Context) error {
 		log.Warnf("Failed to allocate %d IP addresses on an ENI: %v", resourcesToAllocate, err)
 		// Continue to process the allocated IP addresses
 		ipamdErrInc("increaseIPPoolAllocIPAddressesFailed")
+		if containsInsufficientCidrBlocksError(err) {
```

Comment:
how will this help in keep allocating ENI's case? will the IPAMD still keep allocating new ENIs every 2min?

Reply:
For existing ENIs, while increasing the dataStore pool we always land here - https://github.com/aws/amazon-vpc-cni-k8s/blob/867e3be9181989413e487f5074f6a40eb1b4b5b1/pkg/ipamd/ipamd.go#L695 and this will error out for insufficientCidr. 

Only for custom networking, since primary ENI cannot be attached IPs, it falls thru to this condition - https://github.com/aws/amazon-vpc-cni-k8s/blob/867e3be9181989413e487f5074f6a40eb1b4b5b1/pkg/ipamd/ipamd.go#L709 for the first secondary ENI. With this fix, we will allocate an ENI and error out. So next reconciler will tryAssignCidr on the same ENI.